### PR TITLE
Add Manufacturing topic; update upserter to make adding child topics easier

### PR DESCRIPTION
**Pull request by jkachel**
### What are the relevant tickets?

Closes mitodl/hq#5076

### Description (What does it do?)

This adds the Manufacturing topic under Engineering. There are no mappings for this - in the original issue this is to cover a MicroMasters program, and it's got a topic called "Manufacturing" so would be covered by the default by-name mapping for that. 

This also adds a `parent` option to the topics yaml spec so when we're adding/changing sub-topics we can do that without including the entire tree. For example, in https://github.com/mitodl/mit-open/pull/1344 the update adds a topic under Computer Science, and has to include the Computer Science topic to do so. This could potentially be a vector for introducing errors into the topic list - if you grab topic data that's out of date then you'll overwrite changes that have been made with your new stuff. With this code, you can specify a parent UUID so you can just tell it what its parent should be.

### How can this be tested?

Topic: run the migrations for `data_fixtures` and you should see the new Manufacturing topic. It should have Engineering as its parent. It won't have any mappings or an icon.

Upserter updates: Automated tests should pass (there's one specific to this). Additionally, running the migration is a test since it uses the new code. There are changes in the `README-topics.md` too; they should make sense.

### CODE REVIEW

File: data_fixtures/migrations/0008_topics_update_engineering_add_manufacturing.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+def rollback_new_mapping(apps, schema_editor):
+    """Remove the Manufacturing topic."""
+
+    LearningResourceTopic = apps.get_model(
+        "learning_resources", "LearningResourceTopic"
+    )
+    Channel = apps.get_model("channels", "Channel")
+
+    topic = LearningResourceTopic.objects.filter(
+        parent__topic_uuid="952604ab-ae23-45b3-a040-0e5f26fe42df", name="Manufacturing"
```
Comment (mbertrand) [inline data_fixtures/migrations/0008_topics_update_engineering_add_manufacturing.py:40]: Not a big deal, but could this match on the topic uuid instead?

```python
topic = LearningResourceTopic.objects.filter(topic_uuid="4176e385-92c5-4e02-b3cc-4f34d9a4bf40").get()
```

**Review by mbertrand — APPROVED** [review]
Works as expected, just one minor comment. 👍

File: data_fixtures/migrations/0008_topics_update_engineering_add_manufacturing.py (hunk 1 above)
Comment (jkachel) [reply data_fixtures/migrations/0008_topics_update_engineering_add_manufacturing.py]: Fixed this in https://github.com/mitodl/mit-open/pull/1364/commits/17b20709226c96be0cb8da4038467d3aca5ca2b5

**Comment by jkachel** [comment]
@mbertrand just a quick re-review for the rollback change - this shouldn't affect anything but just being cautious.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**